packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +62,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
+      {choices.map(choice => <option key={choice.value} value={choice.value}>{choice.display}</option>)}
```

Comment:
Out of curiosity, how does this change affect the onChange callback? Would we need to add a handler function to return `choice.value` onChange?